src/interfaces/IOptionSettlementEngine.sol

```diff
@@ -59,19 +60,24 @@ interface IOptionSettlementEngine {
     error ExpiredOption(uint256 optionId, uint40 expiry);
 
     /// @notice This option cannot yet be exercised.
-    error ExerciseTooEarly();
+    /// @param optionId Supplied option ID.
+    error ExerciseTooEarly(uint256 optionId);
 
     /// @notice This option has no claims written against it.
-    error NoClaims();
+    /// @param optionId Supplied option ID.
+    error NoClaims(uint256 optionId);
 
     /// @notice This account has no claims.
-    error BalanceTooLow();
+    /// @param claimId Supplied claim ID.
+    error BalanceTooLow(uint256 claimId);
 
     /// @notice This claimId has already been claimed.
-    error AlreadyClaimed();
+    /// @param claimId Supplied claim ID.
+    error AlreadyClaimed(uint256 claimId);
 
     /// @notice You can't claim before expiry.
-    error ClaimTooSoon();
+    /// @param claimId Supplied claim ID.
+    error ClaimTooSoon(uint256 claimId);
```

Comment:
Would it be useful to emit the expiry timestamp here in addition to the claimID? That way, the error emitted would have information necessary for caller to handle.

Reply:
a guiding principle we might consider for errors is to always provide enough information in the error to make it actionable. @0xAlcibiades